docs: adds documentation for Storage Version Migration (#45714)
kubernetes/website: content/en/docs/tasks/manage-kubernetes-objects/storage-version-migration.md

  ```
- Monitor migration of Secrets by checking the `.status` of the StorageVersionMigration.
	A successful migration should have its
`Succeeded` condition set to true. Get the StorageVersionMigration object
as follows:
  ```shell
  kubectl get storageversionmigration.storagemigration.k8s.io/secrets-migration -o yaml
  ```

	The output is similar to:
  ```yaml
  kind: StorageVersionMigration
  apiVersion: storagemigration.k8s.io/v1alpha1
  metadata:
    name: secrets-migration
    uid: 628f6922-a9cb-4514-b076-12d3c178967c
    resourceVersion: '90'
    creationTimestamp: '2024-03-12T20:29:45Z'
  spec:
    resource:
      group: ""
      version: v1
      resource: secrets
  status:
    conditions:
    - type: Running
      status: 'False'
      lastUpdateTime: '2024-03-12T20:29:46Z'
      reason: StorageVersionMigrationInProgress
    - type: Succeeded
      status: 'True'
      lastUpdateTime: '2024-03-12T20:29:46Z'
      reason: StorageVersionMigrationSucceeded
    resourceVersion: '84'
  ```
- [Verify](/docs/tasks/administer-cluster/kms-provider/#verifying-that-the-data-is-encrypted)
  the stored secret is now prefixed with `k8s:enc:aescbc:v1:key2`.

> ## Update the preferred storage schema of the resource while moving from _v1_ to _v2_

Reviewer: How about this?

```suggestion
## Update the preferred storage schema of a CRD
```

schema conversion between them. Before setting v2 as the preferred storage schema
version, it's important to ensure that all existing CRs stored as v1 are migrated to v2.
This migration can be achieved through _Storage Version Migration_ to migrate all CRs from v1 to v2.

- Create a manifest for the CRD, named `test-crd.yaml`, as follows:
  ```yaml
  apiVersion: apiextensions.k8s.io/v1
  kind: CustomResourceDefinition
  metadata:
    name: selfierequests.stable.example.com
  spec:
    group: stable.example.com
    names:
      plural: SelfieRequests
      singular: SelfieRequest
      kind: SelfieRequest
      listKind: SelfieRequestList
    scope: Namespaced
    versions:
    - name: v1
      served: true
      storage: true
      schema:
        openAPIV3Schema:
          type: object
          properties:
            hostPort:
              type: string
    conversion:
      strategy: Webhook
      webhook:
        clientConfig:
          url: https://127.0.0.1:9443/crdconvert
          caBundle: <CABundle info>
      conversionReviewVersions:
      - v1
      - v2
  ```
  Create CRD using kubectl
  ```shell
  kubectl apply -f test-crd.yaml
  ```

> - Create a manifest for an example testcrd. Name the manifest `cr1.yaml` and use these contents:

Reviewer: nit: what is 'testcrd'? this spelling may confuse users.

  ```yaml
  apiVersion: stable.example.com/v1
  kind: SelfieRequest
  metadata:
    name: cr1
    namespace: default
  ```
  Create CR using kubectl
  ```shell
  kubectl apply -f cr1.yaml
  ```
- Verify that CR is written and stored as v1 by getting the object from etcd.
  ```shell
  ETCDCTL_API=3 etcdctl get /kubernetes.io/stable.example.com/testcrds/default/cr1 [...] | hexdump -C
  ```

>   where [...] contains the additional arguments for connecting to the etcd server.

Reviewer: ```suggestion
  where `[...]` contains the additional arguments for connecting to the etcd server.
```

        conversionReviewVersions:
          - v1
          - v2
  ```
  Update CRD using kubectl
  ```shell
  kubectl apply -f test-crd.yaml
  ```
- Create CR resource file with name `cr2.yaml` as follows:
  ```yaml
  apiVersion: stable.example.com/v2
  kind: SelfieRequest
  metadata:
    name: cr2
    namespace: default
  ```
- Create CR using kubectl
  ```shell
  kubectl apply -f cr2.yaml
  ```
- Verify that CR is written and stored as v2 by getting the object from etcd.
  ```shell
  ETCDCTL_API=3 etcdctl get /kubernetes.io/stable.example.com/testcrds/default/cr2 [...] | hexdump -C
  ```
  where [...] contains the additional arguments for connecting to the etcd server.
- Create a StorageVersionMigration manifest named `migrate-crd.yaml`, with the contents as follows:
  ```yaml
  kind: StorageVersionMigration
  apiVersion: storagemigration.k8s.io/v1alpha1
  metadata:
    name: crdsvm
  spec:
    resource:
      group: stable.example.com
      version: v1
      resource: SelfieRequest
  ```
  Create the object using _kubectl_ as follows:
  ```shell
  kubectl apply -f migrate-crd.yaml
  ```
- Monitor migration of secrets using status. Successful migration should have

>   _Succeeded_ condition set to _true_ in the status field. Get the migration resource

Reviewer: ```suggestion
  the `Succeeded` condition set to "True" in the status field. Get the migration resource
```
